tests/suites/cmr/offer_consume.sh

```diff
@@ -0,0 +1,134 @@
+# Exercising the CMR Juju functionality which allows applications to
+# communicate between different models.
+# This test will exercise the following aspects:
+#  - Ensure a user is able to create an offer of an applications' endpoint
+#    including:
+#      - A user is able to consume and relate to the offer
+#      - Workload data successfully provided
+#      - The offer appears in the list-offers output
+#      - The user is able to name the offer
+#      - The user is able to remove the offer
+#  - Ensure an admin can grant a user access to an offer
+#      - The consuming user finds the offer via 'find-offer'
+#
+# The above feature tests will be run on:
+#  - A single controller environment
+run_offer_consume() {
+	# Echo out to ensure nice output to the test suite.
+	echo
+
+	# The following ensures that a bootstrap juju exists.
+	file="${TEST_DIR}/test-offer-consume.log"
+	ensure "model-offer" "${file}"
+
+	echo "Deploy consumed workload and create the offer"
+	juju deploy juju-qa-dummy-source
+	juju offer dummy-source:sink dummy-offer
+
+	wait_for "dummy-source" "$(idle_condition "dummy-source")"
+
+	echo "Deploy workload in consume model"
+	juju add-model "model-consume"
+	juju switch "model-consume"
+	juju deploy juju-qa-dummy-sink
+
+	wait_for "dummy-sink" "$(idle_condition "dummy-sink")"
+
+	echo "Relate workload in consume model with offer"
+	juju --show-log consume "${BOOTSTRAPPED_JUJU_CTRL_NAME}:admin/model-offer.dummy-offer"
+	juju --show-log relate dummy-sink dummy-offer
+	# wait for relation joined before migrate.
+	wait_for "dummy-offer" '.applications["dummy-sink"] | .relations.source[0]'
+
+	echo "Provide config if offered workload and change the status of consumed offer"
+	# Change the dummy-source config for "token" and check that the change
+	# is represented in the consuming model's dummy-sink unit.
+	juju switch "model-offer"
+	juju config dummy-source token=yeah-boi
+	juju switch "model-consume"
+	wait_for "active" '."application-endpoints"["dummy-offer"]."application-status".current'
+
+	echo "Remove offer"
+	# The offer must be removed before model/controller destruction will work.
+	# See discussion under https://bugs.launchpad.net/juju/+bug/1830292.
+	juju switch "model-offer"
+	juju remove-offer "admin/model-offer.dummy-offer" --force -y
```

Comment:
we should test without --force
test that remove fails with relation still present
then remove relation and remove offer